### DISCUSSION TITLE
Input file, when runs fails to give any physical significant data

### DISCUSSION THREAD

**Discussion by soumyaranjanmoharana2005** [terrapower/armi#1799] [Q&A] [answered]
Hey there!
I am a bit new to ARMI and I wanted to run my input file of a reactor with the blueprint of the reactor and fuel shuffler and ran it with the setting file of a terminal app I made based on ARMI. The program runs, but it returns time, power, maxBuF, maxBuI to be zero, although the number of cycles is increasing. Where have I gone wrong?

for context, I've attached the settings file screenshot and the terminal screenshot.

Any help will be appreciated.

Thanks

![Screenshot 2024-07-28 220553](https://github.com/user-attachments/assets/aa07f795-2e77-4988-99da-56424c201ea0)
![Screenshot 2024-07-28 220711](https://github.com/user-attachments/assets/5efd4223-263a-4ef0-a3d9-dd79e371fb8f)

**Comment by jakehader** [comment, marked as answer]
Hi @soumyaranjanmoharana2005 - I am not sure about the contents of the sample fuel handler that you have, but this consistent with how ARMI should work out of box without any flux or depletion solvers. In essence, this is showing that multiple time nodes can be iterated through, but the time, burnup, and power density parameters need to be established by calling external plugins. In this example, there are no plugins connected to the "application" that is being run and therefore the initial states of the parameters are not changing.

**Reply by soumyaranjanmoharana2005** [reply]
@jakehader I'm not sure of any external plugins you're talking about. Could you give me some examples?

**Reply by jakehader** [reply]
This is a useful part of the manual:

https://terrapower.github.io/armi/developer/making_armi_based_apps.html

https://terrapower.github.io/armi/tutorials/making_your_first_app.html#armi-make-first-app

Here is an example plugin that we have open sourced to interact with the OpenMC code, but note that this is still in development, may not be fully featured and would require the OpenMC code itself as a dependency. 

https://github.com/terrapower/armicontrib-openmc

**Comment by soumyaranjanmoharana2005** [comment]
@jakehader This is an excerpt from powershell while running the settings file.


[impt] Cycle 1 Summary:
[info] Core Average Burnup: 0.0
       Idealized Outlet Temperature 0.0
       End of Cycle 01. Timestamp: Mon Jul 29 14:38:36 2024

       
Do I need to setup any burnup or anything else in the setting?

And for the external modules, ARMI can independently do the neutronics calculations, right?

If yes, please tell me if in the code, the reactor is not starting or the reactor has started but there is an error in returning the data?

Any help will be appreciated.

**Reply by jakehader** [reply]
The ARMI framework does not ship with any physics solvers, but rather provides the building blocks for someone or an organization to attach their own custom plugins. I linked a part of the docs in my comment above for more background information and an example application. 

If you're interested in burnup then you'll need both a flux solver and a depletion code. There are many codes for this, but this is a useful list: https://github.com/paulromano/awesome-nuclear
